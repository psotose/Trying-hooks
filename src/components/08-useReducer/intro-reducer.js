const initialState = [{
  id: 1,
  todo: 'Comprar pan',
  done: false
}];

const myFirstTodoReducer = (state = initialState, action) => {

  if ( action?.type === 'add' ){
    return [ ...state, action.payload ];
  }
  return state;
}

let todos = myFirstTodoReducer();

const newTodo = [{
  id: 2,
  todo: 'Comprar leche',
  done: false
}];

const addTodoAction = {
  type: 'add',
  payload: newTodo,
}

todos = myFirstTodoReducer( todos, addTodoAction);

console.log(todos);