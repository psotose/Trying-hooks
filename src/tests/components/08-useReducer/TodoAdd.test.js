import React from 'react';
import { TodoAdd } from '../../../components/08-useReducer/TodoAdd';
import { shallow } from 'enzyme';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Testing <TodoAdd />', () => {

  const handleAddTodo = jest.fn();

  const wrapper = shallow(
    <TodoAdd 
      handleAddTodo= { handleAddTodo }
    />
  )
  test('should render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should not call handleAddTodo', () => {

    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault(){} });

    expect( handleAddTodo ).toHaveBeenCalledTimes(0);
  });

  test('should call handleAddTodo function', () => {

    const value = 'Learn React';
    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description',
      }
    });

    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault(){} });
    expect( handleAddTodo ).toHaveBeenCalledTimes(1);

    expect( handleAddTodo ).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false
    });
    //esta para evaluar que se ejecuta el reset
    expect( wrapper.find('input').prop('value') ).toBe('');
  });

});