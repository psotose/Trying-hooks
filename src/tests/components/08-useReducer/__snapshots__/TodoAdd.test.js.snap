// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TodoAdd /> should render correctly 1`] = `
<Fragment>
  <h4>
    Agregar ToDo
  </h4>
  <hr />
  <form
    className="d-grid"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 d-grid"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
